webui/react/src/pages/WorkspaceList/WorkspaceActionDropdown.tsx

```diff
@@ -44,9 +43,15 @@ const WorkspaceActionDropdown: React.FC<Props> = ({
     modalOpen: openWorkspaceEdit,
   } = useModalWorkspaceEdit({ onClose: onComplete, workspace });
 
-  const userHasPermissions = useMemo(() => {
-    return curUser?.isAdmin || curUser?.id === workspace.userId;
-  }, [ curUser?.id, curUser?.isAdmin, workspace.userId ]);
+  const { canDeleteWorkspace, canModifyWorkspace } = usePermissions();
+
+  const canDelete = useMemo(() => {
```

Comment:
As above I don't know that we really need these `useMemo`s since these are trivial computations.